pkg/controller/operators/olm/operator.go

```diff
@@ -429,9 +454,67 @@ func (a *Operator) syncClusterServiceVersion(obj interface{}) (syncError error)
 		}
 		syncError = fmt.Errorf("error transitioning ClusterServiceVersion: %s and error updating CSV status: %s", syncError, updateErr)
 	}
+
+	// Check if we need to do any copying / annotation for the operatorgroup
+	if err := a.copyCsvToTargetNamespace(updatedCSV, operatorGroup); err != nil {
+		logger.WithError(err).Info("couldn't copy CSV to target namespaces")
+	}
+
+	// Ensure operator has access to targetnamespaces
+	if err := a.ensureRBACInTargetNamespace(updatedCSV, operatorGroup); err != nil {
+		logger.WithError(err).Info("couldn't ensure RBAC in target namespaces")
+	}
+
+	// Ensure cluster roles exist for using provided apis
+	if err := a.ensureClusterRolesForCSV(updatedCSV, operatorGroup); err != nil {
+		logger.WithError(err).Info("couldn't ensure clusterroles for provided api types")
+	}
+
 	return
 }
 
+// operatorGroupForCSV returns the operatorgroup for the CSV only if the CSV is active one in the group
+func (a *Operator) operatorGroupForActiveCSV(logger *logrus.Entry, csv *v1alpha1.ClusterServiceVersion) *v1alpha2.OperatorGroup {
+	annotations := csv.GetAnnotations()
+
+	// not part of a group yet
+	if annotations == nil {
+		logger.Info("not part of any operatorgroup, no annotations")
+		return nil
+	}
+
+	// not in the operatorgroup namespace
+	if annotations[operatorGroupNamespaceAnnotationKey] != csv.GetNamespace() {
+		logger.Info("not in operatorgroup namespace, skipping")
+		return nil
+	}
+
+	operatorGroupName, ok := annotations[operatorGroupAnnotationKey]
+
+	// no operatorgroup annotation
+	if !ok {
+		logger.Info("no operatorgroup annotation")
+		return nil
+	}
+
+	logger = logger.WithField("operatorgroup", operatorGroupName)
+
+	operatorGroup, err := a.lister.OperatorsV1alpha2().OperatorGroupLister().OperatorGroups(csv.GetNamespace()).Get(operatorGroupName)
+	// operatorgroup not found
+	if err != nil {
+		logger.Info("operatorgroup not found")
+		return nil
+	}
+
+	// target namespaces don't match
+	if annotations[operatorGroupTargetsAnnotationKey] != strings.Join(operatorGroup.Status.Namespaces, ",") {
```

Comment:
Would this scenario only occur if an operator group was modified after annotations had been previously correct?